src/solution/solution.jl

```diff
@@ -53,66 +59,70 @@ function solution(model::JuMP.Model)::OrderedDict
+            r.name => [
+                value(model[:reserve_shortfall][sc.name, r.name, t]) for
                 t in 1:instance.time
-            ] for g in r.units
-        ) for r in instance.reserves if r.type == "spinning"
-    )
-    sol["Spinning reserve shortfall (MW)"] = OrderedDict(
-        r.name => [
-            value(model[:reserve_shortfall][r.name, t]) for
-            t in 1:instance.time
-        ] for r in instance.reserves if r.type == "spinning"
-    )
-    sol["Up-flexiramp (MW)"] = OrderedDict(
-        r.name => OrderedDict(
-            g.name => [
-                value(model[:upflexiramp][r.name, g.name, t]) for
+            ] for r in sc.reserves if r.type == "spinning"
+        )
+        sol[sc.name]["Up-flexiramp (MW)"] = OrderedDict(
+            r.name => OrderedDict(
+                g.name => [
+                    value(model[:upflexiramp][sc.name, r.name, g.name, t]) for
+                    t in 1:instance.time
+                ] for g in r.units
+            ) for r in sc.reserves if r.type == "up-frp"
+        )
+        sol[sc.name]["Up-flexiramp shortfall (MW)"] = OrderedDict(
+            r.name => [
+                value(model[:upflexiramp_shortfall][sc.name, r.name, t]) for
                 t in 1:instance.time
-            ] for g in r.units
-        ) for r in instance.reserves if r.type == "flexiramp"
-    )
-    sol["Up-flexiramp shortfall (MW)"] = OrderedDict(
-        r.name => [
-            value(model[:upflexiramp_shortfall][r.name, t]) for
-            t in 1:instance.time
-        ] for r in instance.reserves if r.type == "flexiramp"
-    )
-    sol["Down-flexiramp (MW)"] = OrderedDict(
-        r.name => OrderedDict(
-            g.name => [
-                value(model[:dwflexiramp][r.name, g.name, t]) for
+            ] for r in sc.reserves if r.type == "up-frp"
+        )
+        sol[sc.name]["Down-flexiramp (MW)"] = OrderedDict(
+            r.name => OrderedDict(
+                g.name => [
+                    value(model[:dwflexiramp][sc.name, r.name, g.name, t]) for
+                    t in 1:instance.time
+                ] for g in r.units
+            ) for r in sc.reserves if r.type == "down-frp"
+        )
+        sol[sc.name]["Down-flexiramp shortfall (MW)"] = OrderedDict(
+            r.name => [
+                value(model[:dwflexiramp_shortfall][sc.name, r.name, t]) for
                 t in 1:instance.time
-            ] for g in r.units
-        ) for r in instance.reserves if r.type == "flexiramp"
-    )
-    sol["Down-flexiramp shortfall (MW)"] = OrderedDict(
-        r.name => [
-            value(model[:upflexiramp_shortfall][r.name, t]) for
-            t in 1:instance.time
-        ] for r in instance.reserves if r.type == "flexiramp"
-    )
+            ] for r in sc.reserves if r.type == "down-frp"
+        )
+    end
+    length(keys(sol)) > 1 ? nothing : sol = Dict(sol_key => sol_val for scen_key in keys(sol) for (sol_key, sol_val) in sol[scen_key])
```

Comment:
This is a bit cryptic. The following would be more clear:
```julia
if length(instance.scenarios) == 1)
    return first(values(sol))
else
    return sol
end

Reply:
Done